os/arch/arm/src/armv7-m/up_assert.c

```diff
@@ -440,11 +443,27 @@ static void _up_assert(int errorcode)
 		request_msg.cmd = BINMGR_FAULT;
 		request_msg.pid = getpid();
 
+		if (current_regs) {
+			tcb = sched_self();
+			sched_removereadytorun(tcb);
+#if CONFIG_RR_INTERVAL > 0
+			tcb->timeslice = 0;
+#endif
+			tcb->sched_priority = SCHED_PRIORITY_MIN;
+			sched_addreadytorun(tcb);
+		}
+
 		/* Send a message to fault manager with pid of the faulty task */
 		ret = mq_send(binmgr_mq, (const char *)&request_msg, sizeof(binmgr_request_t), BINMGR_FAULT_PRIO);
 		DEBUGASSERT(ret == 0);
+
+		if (current_regs) {
+			tcb = sched_self();
+			current_regs = tcb->xcp.regs;
+		}
```

Comment:
After sending a message to binary manager, this will not be called anymore. Because binary manager will terminate all of tasks which belong to asserted binary. Why do we need this?

Reply:
1. binary manager will be scheduled after the current assert function returns/exit.
2. The code is needed as it is setting the "current_regs" to the next readytorun task not the faulty asserted binary.